### ISSUE TITLE
lr_scheduler.step() should pass epoch variable to be consistent with PyTorch _LRScheduler

### ISSUE THREAD

**Issue by thinline72** [PyTorchLightning/pytorch-lightning#1333]
PyTorch `_LRScheduler` base class for schedulers supports passing `epoch` parameter to the `lr_scheduler.step()` [function](https://github.com/pytorch/pytorch/blob/master/torch/optim/lr_scheduler.py#L106).

But when Lightning training loop [updates learning rates](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/trainer/training_loop.py#L767), it doesn't pass the current epoch index. That leads to a wrong behaviour of those shedulers that relies on `epoch` param.

Some schedulers handle that under the hood, like [CosineAnnealingWarmRestarts](https://github.com/pytorch/pytorch/blob/master/torch/optim/lr_scheduler.py#L898) from PyTorch that [increment last epoch index](https://github.com/pytorch/pytorch/blob/master/torch/optim/lr_scheduler.py#L976-L977) if epoch isn't provided. In my case I have a custom scheduler that updates learning rate after each batch via `"interval": "step"` and it also uses current epoch index. I can create a workaround for sure, but thought it'd be nice to get it fixed in the Pytorch Lightning too. Plus it wasn't obvious for me until I logged lr into tensorboard.

**Comment by thinline72** [comment]
Actually, `_LRScheduler` also handles that by [incrementing the last epoch index](https://github.com/pytorch/pytorch/blob/master/torch/optim/lr_scheduler.py#L139-L141) if epoch isn't provided.

Then it's an issue for all schedulers that update lr after each batch (like with `"interval": "step"`) and relies on that `self. last_epoch`. As it increments `self. last_epoch` after each batch, it basically becomes last step index.

**Comment by thinline72** [comment]
Oh, I see that PyTorch has a [deprecation warning](https://github.com/pytorch/pytorch/blob/master/torch/optim/lr_scheduler.py#L143) for the epoch param now. So probably it doesn't make sense to fix anymore, I'll close the issue. 

At least it'll be available in search if someone get stuck like me today :)
